Marlin/ultralcd.cpp

```diff
@@ -1013,17 +1022,23 @@ void lcd_cooldown() {
       current_position[Z_AXIS] = MESH_HOME_SEARCH_Z;
       plan_set_position(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], current_position[E_AXIS]);
       lcd_goto_menu(_lcd_level_goto_next_point, true);
+      lcdDrawUpdate = LCDVIEW_CALL_REDRAW_NEXT;
     }
```

Comment:
Same here, because `lcd_goto_menu` sets `lcdDrawUpdate` to `LCDVIEW_CLEAR_CALL_REDRAW`. Was that not working? Or was that unnecessary?

You can see here one of the concerns I had that led to my "hack." If we call `lcd_goto_menu` or set `currentMenu` in the first time through the **graphical display drawing loop** then for the rest of the loop it's drawing the next screen. It would be better to keep calling the same handler (only to draw) for the entire drawing loop, and only change to the next menu after that.

The problem I discovered with my "hack" introducing `nextMenu` was that sometimes you do want the current menu to change right away to avoid running the same logic more than once (for example, the "if axis homed then goto next screen" logic that calls `lcd_goto_menu` should only have run when `nextMenu` is null).

Anyway, I've thrown all that away. Ultimately —as you say— simplicity must win the day. But I also want to avoid spending electrons and heat continually updating the display (even if only for a few seconds) if it's not needed. All those electrons start to add up — and we burn coal here.